x/auth/vesting/msg_server.go

```diff
@@ -44,6 +44,19 @@ func (s msgServer) CreateVestingAccount(goCtx context.Context, msg *types.MsgCre
 		return nil, err
 	}
 
+	if msg.EndTime <= 0 {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "invalid end time")
+	}
+
+	if msg.EndTime <= msg.StartTime {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "invalid start and end time (must be start < end)")
+	}
```

Comment:
since these are stateless checks, ideally we put these in the validate basic function

Reply:
Yeah, will add these there instead.

Reply:
Note that these are already checkec in the corresponding ValidateBasic: https://github.com/celestiaorg/cosmos-sdk/blob/55054282d2df794d9a5fe2599ea25473379ebc3d/x/auth/vesting/types/vesting_account.go#L276-L282